### DISCUSSION TITLE
Editing metadata for external libraries

### DISCUSSION THREAD

**Discussion by yds0yds** [immich-app/immich#6133] [Q&A]
Is it possible to edit meta data for images in external libraries?  I have a bunch of scanned photos that I have in an external library. I'd like to be able to manually add date/times and locations to them for Immich searching.  I understand from documentation that files in external libraries are not touched (which is good), but I'd like to be able to add the metadata somehow and was hoping it would just get stored elsewhere.

**Comment by aisbergde** [comment]
I edit meta data with digiKam (maging geo tags, removing duplicates, ...). You could use any other software. And because it's an external library for immich, all my external changes are detected and accepted by immich. This works great. You just need immich to say to rescan. Some users are doing this even automatically (not yet tested, manual rescan works fine for me until now)

**Comment by yds0yds** [comment]
Thanks.  I have a lot of scanned files.  I was hoping to upload them to my PikaPod instance (using SFTP to an external folder) and then gradually, over time, edit the metadata.  That way I could share the files with family, and gradually get all the metadata correct (which my family don't care about, but I do!).

**Reply by aisbergde** [reply]
And what prevents you to do it this way? Use it in an external library, edit over time, refresh (rescan) and it should work fine.

**Reply by AlleNPooh** [reply]
这个特性确实很需要，但是希望能够选择是否修改源文件的metadata，这样的话能满足更多人的需求。

**Comment by JKleinrath** [comment]
Thanks @aisbergde, good to know that this works with rescans. I also like the idea of being able to do that in immich, so I don't have to go back and forth. (For me it is not scanned images but some pictures with wrong metadata.)
